programs/ui-wrapper/src/processors/settle_funds.rs

```diff
@@ -0,0 +1,188 @@
+    let manifest_program: Program =
+        Program::new(next_account_info(account_iter)?, &manifest::id())?;
+    let platform_token_account: &AccountInfo = next_account_info(account_iter)?;
+    let referrer_token_account: Result<&AccountInfo, ProgramError> =
+        next_account_info(account_iter);
+
+    check_signer(&wrapper_state, owner.key);
+    let market_info_index: DataIndex = get_market_info_index_for_market(&wrapper_state, market.key);
+
+    // Do an initial sync to get all existing orders and balances fresh. This is
+    // needed for modifying user orders for insufficient funds.
+    sync_fast(&wrapper_state, &market, market_info_index)?;
+
+    let mut wrapper_data: RefMut<&mut [u8]> = wrapper_state.info.try_borrow_mut_data()?;
+    let mut wrapper: DynamicAccount<&mut ManifestWrapperUserFixed, &mut [u8]> =
+        get_mut_dynamic_account(&mut wrapper_data);
+
+    let market_info: &mut MarketInfo =
+        get_mut_helper::<RBNode<MarketInfo>>(&mut wrapper.dynamic, market_info_index)
+            .get_mut_value();
+
+    let WrapperSettleFundsParams {
+        fee_mantissa,
+        platform_fee_percent,
+    } = WrapperSettleFundsParams::try_from_slice(data)?;
+
+    let fee_amount =
+        (market_info.quote_volume_unpaid.as_u64() as u128 * fee_mantissa as u128) / FEE_DENOMINATOR;
+    let effective_quote_volume =
+        QuoteAtoms::new((fee_amount * FEE_DENOMINATOR / fee_mantissa as u128) as u64);
+    market_info.quote_volume_unpaid -= effective_quote_volume;
+
+    let MarketInfo {
+        base_balance,
+        quote_balance,
+        ..
+    } = market_info.clone();
+
+    drop(wrapper_data);
+
+    // settle base
+    invoke(
+        &withdraw_instruction(
+            market.key,
+            owner.key,
+            mint_base.key,
+            base_balance.as_u64(),
+            trader_token_account_base.key,
+            *token_program_base.key,
+        ),
+        &[
+            market.info.clone(),
+            owner.info.clone(),
+            mint_base.clone(),
+            trader_token_account_base.clone(),
+            vault_base.clone(),
+            token_program_base.clone(),
+            manifest_program.info.clone(),
+        ],
+    )?;
+
+    // settle quote
+    invoke(
+        &withdraw_instruction(
+            market.key,
+            owner.key,
+            mint_quote.key,
+            quote_balance.as_u64(),
+            trader_token_account_quote.key,
+            *token_program_quote.key,
+        ),
+        &[
+            market.info.clone(),
+            owner.info.clone(),
+            mint_quote.clone(),
+            trader_token_account_quote.clone(),
+            vault_quote.clone(),
+            token_program_quote.clone(),
+            manifest_program.info.clone(),
+        ],
+    )?;
+
+    // pay fees
+
+    if *vault_quote.owner == spl_token_2022::id() {
+        unimplemented!("token2022 not yet supported")
```

Comment:
with 22 and transfer fees, should we make withdrawing into the fee accounts directly?

Reply:
yeah should do